galasa-ui/src/tests/components/TestRunDetails.test.tsx

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import React from 'react';
+import { render, screen, act } from '@testing-library/react';
+import TestRunDetails from '@/components/test-runs/TestRunDetails';
+
+// Deferred helper (unchanged)
+function deferred<T>() {
+  let resolve!: (value: T) => void;
+  let reject!: (reason?: any) => void;
+  const promise = new Promise<T>((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return { promise, resolve, reject };
+}
```

Comment:
Not clear what `deferred<T>()` does, and the comment doesn't explain much!

Is this needed?